### PR TITLE
AnnotationBear: Allow external coalang

### PR DESCRIPTION
**Pull request by abhsag24**
coala has a new external coalang feature, which is
useful for testing and also gives users an option to use their own
coalang files.

### CODE REVIEW

**Comment by abhsag24** [comment]
@fneu review :)

File: requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Use >= for development versions so that source builds always work
-coala>=0.7.0.dev20160602092230
+coala>=0.7.0.dev20160614070242
```
Comment (Adrianzatreanu) [inline requirements.txt]: not so sure, but shouldnt the coala update be in the different commit?

File: tests/general/AnnotationBearTest.py [hunk 2]
```diff
@@ -112,3 +114,14 @@ def test_combined_strings(self):
             self.assertIn(string2, results[0].contents['strings'])
             self.assertIn(string3, results[0].contents['strings'])
             self.assertEqual(results[0].contents['comments'], ())
+
+    def test_external_coalang(self):
+        section3 = Section("")
```
Comment (sils1297) [inline tests/general/AnnotationBearTest.py]: could we not give variables numbers? Don't you already create a section in setUp that you can just fll here? It's mutable.

File: tests/general/AnnotationBearTest.py [hunk 3]
```diff
@@ -112,3 +114,14 @@ def test_combined_strings(self):
...
+        section3.append(Setting('coalang_dir', escape(os.path.join(
+                            os.path.dirname(__file__), 'test_files'), '\\')))
+        section3.append(Setting('language', 'test'))
+        uut = AnnotationBear(section3, Queue())
+        text = ['//comment line 1\n', '"string line 2"']
+        with execute_bear(uut, "F", text) as result:
+            self.assertNotEqual(result[0].contents['strings'], ())
+            self.assertNotEqual(result[0].contents['comments'], ())
```
Comment (sils1297) [inline tests/general/AnnotationBearTest.py]: can't we check the actual positions?

Comment (abhsag24) [inline tests/general/AnnotationBearTest.py]: we can but thats out of the scope of the test right?

File: tests/general/AnnotationBearTest.py (hunk 2 above)
Comment (abhsag24) [inline tests/general/AnnotationBearTest.py]: ohk i'll do that

File: requirements.txt (hunk 1 above)
Comment (abhsag24) [inline requirements.txt]: they're related, dont you think? i can change that if you want though

File: tests/general/AnnotationBearTest.py (hunk 2 above)
Comment (abhsag24) [inline tests/general/AnnotationBearTest.py]: but wouldn't that change it for further tests? if i use the section in the setup?

**Comment by fneu** [comment]
ack https://github.com/coala-analyzer/coala-bears/pull/520/commits/9542e176071cb64a4f7676f98d09609c64bb3245

**Comment by fneu** [comment]
ack

**Comment by sils1297** [comment]
ack 9542e17

**Comment by sils1297** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Comment by abhsag24** [comment]
@rultor merge

**Comment by abhsag24** [comment]
@rultor merge
